batchflow/run_notebook.py

```diff
@@ -0,0 +1,253 @@
+        raise ValueError(
+            """Invalid value for `out_path_db` argument:
+            If `inputs` or `outputs` are provided, then you need to provide `out_path_db` argument."""
+        )
+
+    if isinstance(outputs, str):
+        outputs = [outputs]
+
+    working_dir = './'
+    execute_kwargs = execute_kwargs or {'timeout': -1}
+    executor = ExecutePreprocessor(**execute_kwargs)
+    kernel_manager = KernelManager()
+
+    # Notebook preparation:
+    # Read the notebook, insert cell with inputs, create a notebook for outputs extraction
+    notebook = nbformat.read(path, as_version=4)
+
+    if hide_input:
+        notebook["metadata"].update({"hide_input": True})
+
+    if inputs or outputs:
+        # (Re)create a shelve database
+        shelve.Pickler = Pickler
+        shelve.Unpickler = Unpickler
+
+        with shelve.open(out_path_db) as notebook_db:
+            notebook_db.clear()
+
+        # Code for work with the shelve database from notebooks
+        code_header = f"""\
+                       # Cell inserted during automated execution
+                       import os, shelve
+                       from dill import Pickler, Unpickler
+
+                       shelve.Pickler = Pickler
+                       shelve.Unpickler = Unpickler
+
+                       out_path_db = {repr(out_path_db)}"""
+
+        code_header = dedent(code_header)
+
+    if inputs:
+        # Save `inputs` in the shelve database and create a cell in the `notebook`
+        # for parameters extraction
+        with shelve.open(out_path_db) as notebook_db:
+            notebook_db.update(inputs)
+
+        code = """\n
+               with shelve.open(out_path_db) as notebook_db:
+                   inputs = {**notebook_db}
+
+                   locals().update(inputs)"""
+
+        code = dedent(code)
+        code = code_header + code
+
+        notebook['cells'].insert(inputs_pos, nbformat.v4.new_code_cell(code))
+
+    if outputs:
+        # Create a notebook to extract outputs from the main notebook
+        # The `output_notebook` save locals with preferred names in the shelve database
+        code = f"""
+                # Output dict preparation
+                output = {{}}
+                outputs = {outputs}
+
+                for value_name in outputs:
+                    if value_name in locals():
+                        output[value_name] = locals()[value_name]
+
+                with shelve.open(out_path_db) as notebook_db:
+                    notebook_db['outputs'] = output"""
+
+        code = dedent(code)
+        code = code_header + code
+
+        output_notebook = nbformat.v4.new_notebook(metadata=notebook.metadata)
+        output_notebook['cells'].append(nbformat.v4.new_code_cell(code))
+
+    # Execute the notebook
+    start_time = time.time()
+    try:
+        executor.preprocess(notebook, {'metadata': {'path': working_dir}}, km=kernel_manager)
+        exec_failed = False
```

Comment:
`execution_failed` flag should be created (with `False` value) before `try/except`